crates/revm/src/inspector/gas.rs

```diff
@@ -69,7 +69,15 @@ impl<DB: Database> Inspector<DB> for GasInspector {
         ret: InstructionResult,
         out: Bytes,
     ) -> (InstructionResult, Gas, Bytes) {
-        (ret, remaining_gas, out)
+        match ret {
+            InstructionResult::InvalidFEOpcode | InstructionResult::OpcodeNotFound => {
```

Comment:
About this. This should emulate all `InstructionResult` that will spend all gas.

Check this part:
https://github.com/bluealloy/revm/blob/c54f079a4517c0c58e2808605208278338c24211/crates/interpreter/src/instructions/host.rs#L495-L522

It would be good to have function in `InstructionResult` `fn is_error(&self) -> bool` to check all cases of error kind cases.
All error kind cases are here:
https://github.com/bluealloy/revm/blob/c54f079a4517c0c58e2808605208278338c24211/crates/interpreter/src/instruction_result.rs#L19-L45